### DISCUSSION TITLE
Making exe for Python script with external dependencies

### DISCUSSION THREAD

**Discussion by joshuaahuang** [indygreg/PyOxidizer#612] [Q&A]
I'm trying to convert a Python script using numpy and pandas into a Python executable and I can't seem to figure out how. I'm using the exe.read_package_root and the instruction found here (https://pyoxidizer.readthedocs.io/en/v0.9.0/packaging_python_executable.html) to run and make the .exe, but whenever I try to package files instead of in-memory resources for numpy/pandas (https://pyoxidizer.readthedocs.io/en/v0.9.0/packaging_additional_files.html) the module of my script is no longer found. How can I make the executable work w/ the external dependencies?

**Comment by AceofSpades5757** [comment]
I have also been struggling with this issue, but I haven't been able to find any solutions. Most of the workarounds are giving me errors, but I'm not exactly sure why.

**Reply by AceofSpades5757** [reply]
This was the latest instructions I was trying to get to work.

https://pyoxidizer.readthedocs.io/en/v0.9.0/packaging_additional_files.html#installing-unclassified-files-on-the-filesystem

**Comment by davetapley** [comment]
I just ran in to this too 😬 

I started with minimal example using `requirements.txt` and `config.run_module = "main"` from https://github.com/indygreg/PyOxidizer/issues/246#issuecomment-1272209257, works fine.

Then I introduce `import numpy` and see the `__file__` problem. I workaround that per [here](https://pyoxidizer.readthedocs.io/en/v0.9.0/packaging_additional_files.html#installing-unclassified-files-on-the-filesystem) (via https://github.com/indygreg/PyOxidizer/issues/65#issuecomment-711603334) with:

```py
    policy.set_resource_handling_mode("files")
    policy.resources_location_fallback = "filesystem-relative:lib"

    python_config = dist.make_python_interpreter_config()
    python_config.module_search_paths = ["$ORIGIN/lib"]

# .....

    for resource in exe.pip_download(["numpy==1.24.1"]):
        resource.add_location = "filesystem-relative:lib"
        exe.add_python_resource(resource)
```

But now I get:

```
/ng/oxidizer/build/x86_64-unknown-linux-gnu/debug/install/testapp: No module named main
```

Any tips @indygreg / @safeeval ? 🙏🏻
